common/src/main/kotlin/streams/config/StreamsConfig.kt

```diff
@@ -78,4 +84,17 @@ class StreamsConfig(logService: LogService) : LifecycleAdapter() {
         }
     }
 
+    fun defaultDbName() = this.dbms.getDefaultDbName()
+
+    fun isDefaultDb(dbName: String) = this.defaultDbName() == dbName
+
+    fun isSourceGloballyEnabled() = this.config.getOrDefault(SOURCE_ENABLED, SOURCE_ENABLED_VALUE).toString().toBoolean()
+
+    fun isSourceEnabled(dbName: String) = this.config.getOrDefault("${SOURCE_ENABLED}.from.$dbName", isSourceGloballyEnabled()).toString().toBoolean()
+
+    fun hasProceduresGloballyEnabled() = this.config.getOrDefault(PROCEDURES_ENABLED, PROCEDURES_ENABLED_VALUE).toString().toBoolean()
```

Comment:
this makes me realize that under 4.0 we should make a documentation note: when you enable procedures you do it globally for all databases.  This is OK for now but might be undesirable later.  If people use multi-tenant ops and some people want kafka in/out of database A, it might not make sense that procedures are enabled on database B.

What do you think about making this setting work like the rest - if you use it without a database name it works only for the default database, otherwise you have to enable it per-database?

Reply:
Yes I agree with you, in fact, works already as you want because there is line 97:

https://github.com/neo4j-contrib/neo4j-streams/blob/38ee3379a5d6e6ca7825ce6198c8cf9b22e872e4/common/src/main/kotlin/streams/config/StreamsConfig.kt#L97

Which manages the procedure for each database instance.
If no value is specified for a database instance it inherits the configuration from the global configuration.
Please look at the `configuration.adoc` file for more details about it works

Reply:
great!  My miss.